### DISCUSSION TITLE
Support of EQL searches

### DISCUSSION THREAD

**Discussion by pandvan** [jertel/elastalert2#625] [Q&A] [answered]
Is there any plan to add support to [EQL query](https://www.elastic.co/guide/en/elasticsearch/reference/master/eql.html) language in ElastAlert? It has been recently added to Elasticsearch and allows to search inside event based time series data.
It can be useful to better work with security events when ES is used as a SIEM (Security Information and Event Management).

**Comment by nsano-rururu** [comment, marked as answer]
There are no particular plans. Pull requests are welcome.

**Reply by ThePicpixel** [reply]
I have been looking at what was needed in order to support EQL in ElastAlert.

Basically, what we are doing now is :
- create an ElasticSearch client instance from the elasticsearch library
- parse the yaml rule in order to get the required parameters
- use the search method with the right parameters in order to get the hits

To support EQL, we could use the ElasticSearch.eql object, which is the EQL client that has been initialized when we instanciated the ElasticSearch object. That client has a search method as well.

But the problem is, that the parameters of these 2 different search methods are not the same (see the [documentation](https://elasticsearch-py.readthedocs.io/en/v8.1.1/api.html#module-elasticsearch)).

So from my understanding, adding EQL support will require to rework the way we are working with ElasticSearchClient (maybe by creating another class like EQLClient and using a Factory to create a Client object that we can work with).

But maybe there is something easier ?
